### DISCUSSION TITLE
함수 호출 방식에 따른 this 바인딩을 설명해주세요

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#96] [this]
- 함수 호출 방식에 따른 this 바인딩을 설명해주세요

**Comment by publdaze** [comment]
> - 함수 호출 방식에 따른 this 바인딩을 설명해주세요
- 함수 호출 방식은 크게 일반 함수 호출, 객체의 메서드 호출, 생성자 함수 호출, apply/call/bind를 사용한 간접 호출이 있음
- 일반 함수 호출 - 전역 객체 바인딩, 중첩된 함수도 동일, 일반 함수에서는 객체를 생성하지 않기 때문에 this가 의미가 없어 strict 모드의 경우 undefined
- 객체의 메서드 호출 - 메서드를 호출한 객체 바인딩
- 생성자 함수 호출 - 인스턴스 바인딩
- apply/call/bind를 사용한 간접 호출 - 명시적으로 인수로 바인딩할 객체 전달

**Comment by pipisebastian** [comment]
> * 함수 호출 방식에 따른 this 바인딩을 설명해주세요
- 일반 함수로써 호출할때는 전역 객체를 가리킵니다. 그러나 strict mode에서는 undefined가 바인딩됩니다.
- 메서드로써 호출할때는 메서드를 호출한 객체를 가리킵니다
- 생성자함수로써 호출할때는 새로 만든 인스턴스를 가리킵니다
- call/apply/bind 사용한 간접 호출 -> 명시적으로 this를 binding
